### DISCUSSION TITLE
Can anyone explain why this option set like this necessarily?

### DISCUSSION THREAD

**Discussion by Squirre17** [AFLplusplus/AFLplusplus#1615] [Q&A] [answered]
I'm confused about this option setting when I work with both AFLpp and ASan.
```c
    if (!strstr(x, "abort_on_error=1")) {

      FATAL("Custom ASAN_OPTIONS set without abort_on_error=1 - please fix!");

    }

#ifndef ASAN_BUILD
    if (!getenv("AFL_DEBUG") && !strstr(x, "symbolize=0")) {

      FATAL("Custom ASAN_OPTIONS set without symbolize=0 - please fix!");

    }
```
I want to understand why AFL has to disable Asan's symbolization and call abort instead of exit.

**Comment by vanhauser-thc** [comment, marked as answer]
symbolizing is very time extensive and hurts the speed

**Reply by Squirre17** [reply]
I see. Then what is `abort_on_error` option for?

**Reply by vanhauser-thc** [reply]
to ensure that an asan error crashes the target so afl can detect it. if you set to 0 afl will miss crashes
